### PR TITLE
Add Swagger files for CLU 2022-03-01-preview

### PR DESCRIPTION
**Pull request by ChongTang**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ChongTang** [comment]
Hi @JeffreyRichter, this PR adds CLU's swagger. We added it on top of the swagger of TA. Both services are under the same /language umbrella. However, there are two validation failures which I hope you can help us.

1. The Avocado failure says MISSING_APIS_IN_DEFAULT_TAG, which I don't know how to fix because we do not have a default tag. This is also the first time we have tried to put Swagger files of two services under the same version.
2. The breaking change failure is because we made "error" as required in an error response. Do you know how to override this validation failure?

Thanks for your help.

**Comment by ChongTang** [comment]
The board review issue: https://github.com/Azure/azure-rest-api-specs/issues/17505

**Comment by ChongTang** [comment]
Hi @tjprescott , this PR has the same [MISSING_APIS_IN_DEFAULT_TAG] failure in Avocado. I tried to add analyzeconversations.json in 2022-02-01-preview but it does not help. I want to add analyzeconversations.json only in 2022-03-01-preview. Really appreciate if you can help me out?

**Comment by ChongTang** [comment]
Hi @weidongxu-microsoft  or @tjprescott . This PR has the same MISSING_APIS_IN_DEFAULT_TAG issue. This PR merges CLU's 2022-03-01-preview version into a release/feature branch. Can you help us skip this check? Thanks!

**Comment by weidongxu-microsoft** [comment]
@ChongTang skip CI will only happen before you are ready to merge. I don't think this PR is at this stage yet.

File: specification/cognitiveservices/data-plane/Language/readme.md
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/readme.md:9]: For product names, I'd spell out what they are for customers, IMO:
```suggestion
The current preview release of Conversational Language Understanding and Text Analytics is 2022-03-01-preview.
```

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: This is a breaking change from 2021-11-01-preview, in which you had:

```json
        "parameters": [
          {
            "$ref": "common.json#/parameters/ProjectNameQueryParameter"
          },
          {
            "$ref": "common.json#/parameters/DeploymentNameQueryParameter"
          },
          {
            "$ref": "#/parameters/ConversationAnalysisOptions"
          },
          {
            "$ref": "common.json#/parameters/ApiVersionParameter"
          }
        ],
```

Was this intentional?

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: @annatisch this may mean that we have to conditionally format requests based on the api-version.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: I see the only addition was a `verbose` property to the now-in-body parameters. Was there another reason for moving to an in-body input?

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:159]: Through descriptions in the swagger, I see a mix of capitalized names and none, like "Item" vs. "item" even for the same phrasing. I'd go through this and make everything that isn't either a product name or class name lowercase. If it's a class name, I'd actually use the class (model) name as-is without spaces. It will make the SDK doc comments much less confusing.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:207]: As an example of my previous comment, here you have "conversation **I**tem* while on line 211 below you have "conversation **i**tem".

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:211]: Is this BCP-47? If so, please say that so customers know what valid inputs look like. If not, please also clarify what the expected format is. "language" is rather vague.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:222]: So `ConversationItemBase` will *never* be used for output? If there's any doubt, make this `true` or some SDKs will have future problems.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: This will *never* be used for output? If any doubt, make `true`.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: Actually, I see that `BaseExtractInformation` is returned by `ConversationEnttity`, which is described as, "The entity extraction result of a Conversation project." Since it's a result (a response), `modelAsString` must be `true`. This is true for any model or model property type that can be in a result. This will break SDKs if you ever add a value in a future version.

`modelAsString: false` (or the absence of the property) is only valid for purely input models - and ones that would only ever be input models. If you ever needed to make such a model also output, you'd need to define a new model. Changing `modelAsString` is a breaking change for the SDKs (for most languages, anyway).

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:1190]: Nit: I see both "date/time" and "date time". Would be ideal if you were consistent for doc purposes.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: Or, if you're 100% sure some of these will never change (like units; I'd be skeptical if result discriminators weren't expected to change in time), you can leave this `false`. See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums for details.

**Review by heaths — CHANGES_REQUESTED** [review]
My only concern blocking the PR is about some of the `modelAsString: false` declarations. You should only specify `false` if the model will only ever be input, or if you're 100% certain the discriminator will never change. Is that true for all instances of `false`? Happy to flip to approve if so, though I had a few small nits as well but don't need to block the PR for those.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: Yes. This is a new design and you will find that TA has the same design.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: > This is a breaking change from 2021-11-01-preview, in which you had:
> 
> ```json
>         "parameters": [
>           {
>             "$ref": "common.json#/parameters/ProjectNameQueryParameter"
>           },
>           {
>             "$ref": "common.json#/parameters/DeploymentNameQueryParameter"
>           },
>           {
>             "$ref": "#/parameters/ConversationAnalysisOptions"
>           },
>           {
>             "$ref": "common.json#/parameters/ApiVersionParameter"
>           }
>         ],
> ```
> 
> Was this intentional?

Yes. This is a new design and you will find that TA has the same design.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: > I see the only addition was a `verbose` property to the now-in-body parameters. Was there another reason for moving to an in-body input?

I think this is the new design, projectName and deployment are not in the request boby instead of query parameters.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:39]: I think you mean *are in* the request body, but is still confirmation this was intentional. Just wanted to make sure. We'll definitely need to make some conditional changes to the SDKs.

/cc @annatisch @AhmedLeithy @ZulaMostafa @mshaban-msft

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:159]: Fixed them

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:207]: Fixed

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: I think it should be true. Let me confirm it with @fxs130430 .

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:1190]: Fixed

File: specification/cognitiveservices/data-plane/Language/readme.md
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/readme.md:9]: Fixed

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: Is this the only one you feel confident will never be in a response or will never change otherwise? Please bear in mind that my comments on one line typically apply to any similar instances throughout the file. Apologies if I didn't make that clear.

Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json:687]: This should be true actually.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json]: Confirmed and it should be true.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json]: Should be true. Updated.

File: specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json
Comment (ChongTang) [reply specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json]: Yes it is. Updated the description.

**Review by heaths — CHANGES_REQUESTED** [review]
There's a couple that seem like they should be `true`. Could you double check the following four, though?

These seem like they should be true. The discriminators - especially the first - seem like something that could easily change over time:

* `BaseResolution.resolutionKind`
* `NumericRangeResolution.rangeKind`

These very well could be immutable and never change, but might be good to double check assumptions:

* `DateTimeResolution.dateTimeSubKind`
* `NumberResolution.numberKind`

Apart from that, the changes look good.

**Review by heaths — CHANGES_REQUESTED** [review]
Actually, there's 4 more instances of "language" (or "detectedLanguage") that you might want to document are BCP 47, if, in fact, they are:

* https://github.com/Azure/azure-rest-api-specs/blob/a74a78919147b3a2866b40f6f326d6bcae78908f/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L394
* https://github.com/Azure/azure-rest-api-specs/blob/67e66fc626cb798f29b6026c4ba4a30a3d927e13/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L436
* https://github.com/Azure/azure-rest-api-specs/blob/a74a78919147b3a2866b40f6f326d6bcae78908f/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L561

**Comment by ChongTang** [comment]
> Actually, there's 4 more instances of "language" (or "detectedLanguage") that you might want to document are BCP 47, if, in fact, they are:
> 
> * https://github.com/Azure/azure-rest-api-specs/blob/a74a78919147b3a2866b40f6f326d6bcae78908f/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L394
> * https://github.com/Azure/azure-rest-api-specs/blob/67e66fc626cb798f29b6026c4ba4a30a3d927e13/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L436
> * https://github.com/Azure/azure-rest-api-specs/blob/a74a78919147b3a2866b40f6f326d6bcae78908f/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json#L561

Thanks Heath! I resolve both "language" and "modelAsString" issues. I also searched in the file and resolved all occurrences.

**Review by heaths — APPROVED** [review]

**Pull request merged**
